### PR TITLE
RUN-3265 fixed falsy preload

### PR DESCRIPTION
**Pull request by whyn07m3**
ℹ️   **Description**
Fixes support for falsy preload scripts

➕  **New test**
• [new Window (preload) (falsy value / reset)](https://testing-dashboard.openfin.co/#/app/tests/59944c83d2a02971da3a62d6/edit)

✅  **Test results**
• [Windows 7](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/599492fed2a02971da3a62da)
• [Windows 10](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/5994935bd2a02971da3a62db)

### CODE REVIEW

File: src/browser/convert_options.js [hunk 1]
```diff
@@ -263,10 +263,15 @@ module.exports = {
...
         }
 
-        if (typeof options.preload === 'string') {
-            newOptions.preload = [{ url: options.preload }];
+        const preload = options.preload;
+        if (!preload) {
+            newOptions.preload = []; // for all falsy values
         } else {
-            newOptions.preload = options.preload || [];
+            if (typeof preload === 'string') {
```
Comment (joneit) [inline src/browser/convert_options.js]: Isn't `} else if {` preferred?

File: src/browser/convert_options.js [hunk 2]
```diff
@@ -263,10 +263,15 @@ module.exports = {
             newOptions.permissions = options.permissions;
         }
 
-        if (typeof options.preload === 'string') {
-            newOptions.preload = [{ url: options.preload }];
+        const preload = options.preload;
+        if (!preload) {
+            newOptions.preload = []; // for all falsy values
```
Comment (joneit) [inline src/browser/convert_options.js]: Nice

File: src/renderer/api-decorator.js [hunk 3]
```diff
@@ -427,7 +427,7 @@ limitations under the License.
         let webContentsId = getWebContentsId();
         // Reset state machine values that are set through synchronous handshake between native WebContent lifecycle observers and JS
         options.openfin = true;
-        options.preload = options.preload || winOpts.preload;
+        options.preload = 'preload' in options ? options.preload : winOpts.preload;
```
Comment (joneit) [inline src/renderer/api-decorator.js]: A bit confusing because `winOpts` is not in fact the options for the new child window (which is in `options`)... unless `winOpts` is a standard usage (is it?), let's please take this opportunity to rename `winOpts` to `mainWindowOptions` (or `appOpts`?)

Comment (whyn07m3) [inline src/renderer/api-decorator.js]: @joneit renaming should go in a separate PR

**Comment by HarsimranSingh** [comment]
👍
